zaino-state/src/backends/state.rs

```diff
@@ -853,6 +863,64 @@ impl StateServiceSubscriber {
         }
     }
 
+    /// Fetches all UTXOs from the state service, and returns them in a map.
+    pub(crate) async fn get_txout_set(
+        state: &ReadStateService,
+    ) -> Result<HashMap<transparent::OutPoint, transparent::Output>, StateServiceError> {
+        let mut utxos: HashMap<transparent::OutPoint, transparent::Output> = HashMap::new();
+
+        let mut state = state.clone();
+
+        let zebra_state::ReadResponse::Tip { 0: tip } = state
```

Comment:
Nitpick: Why destructure it this way rather than as a tuple struct?

```suggestion
        let zebra_state::ReadResponse::Tip(tip) = state
```